spring-cloud-kubernetes-test-support/src/main/resources/setup/role.yaml

```diff
@@ -4,6 +4,6 @@ metadata:
   namespace: default
   name: namespace-reader
 rules:
-  - apiGroups: ["", "extensions", "apps"]
-    resources: ["configmaps", "pods", "services", "endpoints", "secrets"]
+  - apiGroups: ["", "extensions", "apps", "discovery.k8s.io"]
```

Comment:
@ryanjbaxter before I even start documentation and explanation on my thought process in this PR, I need your supervisor advice. 

Kubernetes recommends using EndpointSlices instead of Endpoints (think of it a @Deprecated). Now, we can use EndpointSlices as this PR and integration tests prove. The problem is that by switching to EndpointSlices, users would need to change the roles, as this file shows.

So, one hand we have the official k8s recommendation, on the other hand if we do follow it, we kind of introduce a "breaking" change, as users need to adjust to it. 

Of course, we could let users opt-in into EndpointSlices and we could explain in the docs the steps needed for that, but then we would kind of go against what k8s says to do. 

As a side note, there is nothing wrong in using Endpoints, the code still works... wdyt is the correct path forward here? Thank you.

Reply:
As far as I understand the Endpoints API is going to continue to be supported and is not planned to be removed at the moment, correct?

Also it seems like Endpoint Slices is only officially supporting in k8s 1.2.1 right?  If you are using a version prior to that it was still in beta, correct?

I think if we move forward with this, I would make it an opt in feature.  Document how to opt in and the additional configuration that is needed to use it.

Reply:
> As far as I understand the Endpoints API is going to continue to be supported and is not planned to be removed at the moment, correct?

Correct.

> Also it seems like Endpoint Slices is only officially supporting in k8s 1.2.1 right? If you are using a version prior to that it was still in beta, correct?

Yeah, they had alpha support in 1.16, then beta, then stable. 

> I think if we move forward with this, I would make it an opt in feature. Document how to opt in and the additional configuration that is needed to use it.

Excellent! I'll update the PR accordingly.

Reply:
Thank you for your suggestions. I've updated the PR as suggested. Have a good trip next week!